hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java

```diff
@@ -2033,8 +2034,12 @@ public Connection(SocketChannel channel, long lastContact,
       this.isOnAuxiliaryPort = isOnAuxiliaryPort;
       if (addr == null) {
         this.hostAddress = "*Unknown*";
+        this.hostName = this.hostAddress;
```

Comment:
use NetUtils.getHostNameOfIP() and similar for all this. gives us one point of maintenance, testing, upgrading to ipv6 etc